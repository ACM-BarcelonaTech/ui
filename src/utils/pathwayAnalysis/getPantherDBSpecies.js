import pantherDBSpecies from 'utils/pathwayAnalysis/pantherDBSpecies.json';

const formatOuput = (data) => {
  const { genome } = data.search.output.genomes;

  const speciesList = genome.map((species) => ({
    label: species.long_name,
    value: species.short_name,
  }));

  return speciesList;
};

const getPantherDBSpecies = () => {
  const formattedData = formatOuput(pantherDBSpecies);
  return formattedData;
};

export default getPantherDBSpecies;
