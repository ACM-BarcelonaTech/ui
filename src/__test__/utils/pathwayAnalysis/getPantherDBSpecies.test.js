import getPantherDBSpecies from 'utils/pathwayAnalysis/getPantherDBSpecies';

describe('getPantherDBSpecies', () => {
  it('Returns properly formatted', () => {
    const data = getPantherDBSpecies();
    expect(data).toMatchSnapshot();
  });
});
