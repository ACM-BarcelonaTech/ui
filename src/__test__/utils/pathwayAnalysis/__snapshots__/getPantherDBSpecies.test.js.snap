// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPantherDBSpecies Returns properly formatted 1`] = `
Array [
  Object {
    "label": "Homo sapiens",
    "value": "HUMAN",
  },
  Object {
    "label": "Mus musculus",
    "value": "MOUSE",
  },
  Object {
    "label": "Rattus norvegicus",
    "value": "RAT",
  },
  Object {
    "label": "Gallus gallus",
    "value": "CHICK",
  },
  Object {
    "label": "Danio rerio",
    "value": "DANRE",
  },
  Object {
    "label": "Drosophila melanogaster",
    "value": "DROME",
  },
  Object {
    "label": "Caenorhabditis elegans",
    "value": "CAEEL",
  },
  Object {
    "label": "Saccharomyces cerevisiae",
    "value": "YEAST",
  },
  Object {
    "label": "Schizosaccharomyces pombe",
    "value": "SCHPO",
  },
  Object {
    "label": "Dictyostelium discoideum",
    "value": "DICDI",
  },
  Object {
    "label": "Arabidopsis thaliana",
    "value": "ARATH",
  },
  Object {
    "label": "Escherichia coli",
    "value": "ECOLI",
  },
  Object {
    "label": "Emericella nidulans",
    "value": "EMENI",
  },
  Object {
    "label": "Amborella trichopoda",
    "value": "AMBTC",
  },
  Object {
    "label": "Anolis carolinensis",
    "value": "ANOCA",
  },
  Object {
    "label": "Anopheles gambiae",
    "value": "ANOGA",
  },
  Object {
    "label": "Aquifex aeolicus",
    "value": "AQUAE",
  },
  Object {
    "label": "Ashbya gossypii",
    "value": "ASHGO",
  },
  Object {
    "label": "Bacillus cereus",
    "value": "BACCR",
  },
  Object {
    "label": "Bacillus subtilis",
    "value": "BACSU",
  },
  Object {
    "label": "Bacteroides thetaiotaomicron",
    "value": "BACTN",
  },
  Object {
    "label": "Batrachochytrium dendrobatidis",
    "value": "BATDJ",
  },
  Object {
    "label": "Bos taurus",
    "value": "BOVIN",
  },
  Object {
    "label": "Brachypodium distachyon",
    "value": "BRADI",
  },
  Object {
    "label": "Bradyrhizobium diazoefficiens",
    "value": "BRADU",
  },
  Object {
    "label": "Branchiostoma floridae",
    "value": "BRAFL",
  },
  Object {
    "label": "Brassica napus",
    "value": "BRANA",
  },
  Object {
    "label": "Brassica rapa subsp. pekinensis",
    "value": "BRARP",
  },
  Object {
    "label": "Caenorhabditis briggsae",
    "value": "CAEBR",
  },
  Object {
    "label": "Candida albicans",
    "value": "CANAL",
  },
  Object {
    "label": "Canis lupus familiaris",
    "value": "CANLF",
  },
  Object {
    "label": "Capsicum annuum",
    "value": "CAPAN",
  },
  Object {
    "label": "Chlamydia trachomatis",
    "value": "CHLTR",
  },
  Object {
    "label": "Chlamydomonas reinhardtii",
    "value": "CHLRE",
  },
  Object {
    "label": "Chloroflexus aurantiacus",
    "value": "CHLAA",
  },
  Object {
    "label": "Ciona intestinalis",
    "value": "CIOIN",
  },
  Object {
    "label": "Citrus sinensis",
    "value": "CITSI",
  },
  Object {
    "label": "Clostridium botulinum",
    "value": "CLOBH",
  },
  Object {
    "label": "Coxiella burnetii",
    "value": "COXBU",
  },
  Object {
    "label": "Cryptococcus neoformans",
    "value": "CRYNJ",
  },
  Object {
    "label": "Cucumis sativus",
    "value": "CUCSA",
  },
  Object {
    "label": "Daphnia pulex",
    "value": "DAPPU",
  },
  Object {
    "label": "Deinococcus radiodurans",
    "value": "DEIRA",
  },
  Object {
    "label": "Dictyoglomus turgidum",
    "value": "DICTD",
  },
  Object {
    "label": "Dictyostelium purpureum",
    "value": "DICPU",
  },
  Object {
    "label": "Entamoeba histolytica",
    "value": "ENTHI",
  },
  Object {
    "label": "Equus caballus",
    "value": "HORSE",
  },
  Object {
    "label": "Erythranthe guttata",
    "value": "ERYGU",
  },
  Object {
    "label": "Eucalyptus grandis",
    "value": "EUCGR",
  },
  Object {
    "label": "Felis catus",
    "value": "FELCA",
  },
  Object {
    "label": "Fusobacterium nucleatum",
    "value": "FUSNN",
  },
  Object {
    "label": "Geobacter sulfurreducens",
    "value": "GEOSL",
  },
  Object {
    "label": "Giardia intestinalis",
    "value": "GIAIC",
  },
  Object {
    "label": "Gloeobacter violaceus",
    "value": "GLOVI",
  },
  Object {
    "label": "Glycine max",
    "value": "SOYBN",
  },
  Object {
    "label": "Gorilla gorilla gorilla",
    "value": "GORGO",
  },
  Object {
    "label": "Gossypium hirsutum",
    "value": "GOSHI",
  },
  Object {
    "label": "Haemophilus influenzae",
    "value": "HAEIN",
  },
  Object {
    "label": "Halobacterium salinarum",
    "value": "HALSA",
  },
  Object {
    "label": "Helianthus annuus",
    "value": "HELAN",
  },
  Object {
    "label": "Helicobacter pylori",
    "value": "HELPY",
  },
  Object {
    "label": "Hordeum vulgare subsp. vulgare",
    "value": "HORVV",
  },
  Object {
    "label": "Ixodes scapularis",
    "value": "IXOSC",
  },
  Object {
    "label": "Juglans regia",
    "value": "JUGRE",
  },
  Object {
    "label": "Korarchaeum cryptofilum",
    "value": "KORCO",
  },
  Object {
    "label": "Lactuca sativa",
    "value": "LACSA",
  },
  Object {
    "label": "Leishmania major",
    "value": "LEIMA",
  },
  Object {
    "label": "Leptospira interrogans",
    "value": "LEPIN",
  },
  Object {
    "label": "Listeria monocytogenes",
    "value": "LISMO",
  },
  Object {
    "label": "Macaca mulatta",
    "value": "MACMU",
  },
  Object {
    "label": "Manihot esculenta",
    "value": "MANES",
  },
  Object {
    "label": "Medicago truncatula",
    "value": "MEDTR",
  },
  Object {
    "label": "Methanocaldococcus jannaschii",
    "value": "METJA",
  },
  Object {
    "label": "Methanosarcina acetivorans",
    "value": "METAC",
  },
  Object {
    "label": "Monodelphis domestica",
    "value": "MONDO",
  },
  Object {
    "label": "Monosiga brevicollis",
    "value": "MONBE",
  },
  Object {
    "label": "Musa acuminata subsp. malaccensis",
    "value": "MUSAM",
  },
  Object {
    "label": "Mycobacterium tuberculosis",
    "value": "MYCTU",
  },
  Object {
    "label": "Neisseria meningitidis serogroup b",
    "value": "NEIMB",
  },
  Object {
    "label": "Nelumbo nucifera",
    "value": "NELNU",
  },
  Object {
    "label": "Nematostella vectensis",
    "value": "NEMVE",
  },
  Object {
    "label": "Neosartorya fumigata",
    "value": "ASPFU",
  },
  Object {
    "label": "Neurospora crassa",
    "value": "NEUCR",
  },
  Object {
    "label": "Nicotiana tabacum",
    "value": "TOBAC",
  },
  Object {
    "label": "Nitrosopumilus maritimus",
    "value": "NITMS",
  },
  Object {
    "label": "Ornithorhynchus anatinus",
    "value": "ORNAN",
  },
  Object {
    "label": "Oryza sativa",
    "value": "ORYSJ",
  },
  Object {
    "label": "Oryzias latipes",
    "value": "ORYLA",
  },
  Object {
    "label": "Ostreococcus tauri",
    "value": "OSTTA",
  },
  Object {
    "label": "Pan troglodytes",
    "value": "PANTR",
  },
  Object {
    "label": "Paramecium tetraurelia",
    "value": "PARTE",
  },
  Object {
    "label": "Phaeosphaeria nodorum",
    "value": "PHANO",
  },
  Object {
    "label": "Phoenix dactylifera",
    "value": "PHODC",
  },
  Object {
    "label": "Physcomitrella patens",
    "value": "PHYPA",
  },
  Object {
    "label": "Phytophthora ramorum",
    "value": "PHYRM",
  },
  Object {
    "label": "Plasmodium falciparum",
    "value": "PLAF7",
  },
  Object {
    "label": "Populus trichocarpa",
    "value": "POPTR",
  },
  Object {
    "label": "Pristionchus pacificus",
    "value": "PRIPA",
  },
  Object {
    "label": "Prunus persica",
    "value": "PRUPE",
  },
  Object {
    "label": "Pseudomonas aeruginosa",
    "value": "PSEAE",
  },
  Object {
    "label": "Puccinia graminis",
    "value": "PUCGT",
  },
  Object {
    "label": "Pyrobaculum aerophilum",
    "value": "PYRAE",
  },
  Object {
    "label": "Rhodopirellula baltica",
    "value": "RHOBA",
  },
  Object {
    "label": "Ricinus communis",
    "value": "RICCO",
  },
  Object {
    "label": "Salmonella typhimurium",
    "value": "SALTY",
  },
  Object {
    "label": "Sclerotinia sclerotiorum",
    "value": "SCLS1",
  },
  Object {
    "label": "Selaginella moellendorffii",
    "value": "SELML",
  },
  Object {
    "label": "Setaria italica",
    "value": "SETIT",
  },
  Object {
    "label": "Shewanella oneidensis",
    "value": "SHEON",
  },
  Object {
    "label": "Solanum lycopersicum",
    "value": "SOLLC",
  },
  Object {
    "label": "Solanum tuberosum",
    "value": "SOLTU",
  },
  Object {
    "label": "Sorghum bicolor",
    "value": "SORBI",
  },
  Object {
    "label": "Spinacia oleracea",
    "value": "SPIOL",
  },
  Object {
    "label": "Staphylococcus aureus",
    "value": "STAA8",
  },
  Object {
    "label": "Streptococcus pneumoniae",
    "value": "STRR6",
  },
  Object {
    "label": "Streptomyces coelicolor",
    "value": "STRCO",
  },
  Object {
    "label": "Strongylocentrotus purpuratus",
    "value": "STRPU",
  },
  Object {
    "label": "Sulfolobus solfataricus",
    "value": "SACS2",
  },
  Object {
    "label": "Sus scrofa",
    "value": "PIG",
  },
  Object {
    "label": "Synechocystis",
    "value": "SYNY3",
  },
  Object {
    "label": "Thalassiosira pseudonana",
    "value": "THAPS",
  },
  Object {
    "label": "Theobroma cacao",
    "value": "THECC",
  },
  Object {
    "label": "Thermococcus kodakaraensis",
    "value": "THEKO",
  },
  Object {
    "label": "Thermodesulfovibrio yellowstonii",
    "value": "THEYD",
  },
  Object {
    "label": "Thermotoga maritima",
    "value": "THEMA",
  },
  Object {
    "label": "Tribolium castaneum",
    "value": "TRICA",
  },
  Object {
    "label": "Trichomonas vaginalis",
    "value": "TRIVA",
  },
  Object {
    "label": "Trichoplax adhaerens",
    "value": "TRIAD",
  },
  Object {
    "label": "Triticum aestivum",
    "value": "WHEAT",
  },
  Object {
    "label": "Trypanosoma brucei",
    "value": "TRYB2",
  },
  Object {
    "label": "Ustilago maydis",
    "value": "USTMA",
  },
  Object {
    "label": "Vibrio cholerae",
    "value": "VIBCH",
  },
  Object {
    "label": "Vitis vinifera",
    "value": "VITVI",
  },
  Object {
    "label": "Xanthomonas campestris",
    "value": "XANCP",
  },
  Object {
    "label": "Xenopus tropicalis",
    "value": "XENTR",
  },
  Object {
    "label": "Yarrowia lipolytica",
    "value": "YARLI",
  },
  Object {
    "label": "Yersinia pestis",
    "value": "YERPE",
  },
  Object {
    "label": "Zea mays",
    "value": "MAIZE",
  },
  Object {
    "label": "Zostera marina",
    "value": "ZOSMR",
  },
  Object {
    "label": "helobdella robusta",
    "value": "HELRO",
  },
  Object {
    "label": "lepisosteus oculatus",
    "value": "LEPOC",
  },
  Object {
    "label": "mycoplasma genitalium",
    "value": "MYCGE",
  },
]
`;
